sdk/typescript/src/signers/txn-data-serializers/local-txn-data-serializer.ts

```diff
@@ -89,6 +90,31 @@ export class LocalTxnDataSerializer implements TxnDataSerializer {
     }
   }
 
+  async newPay(
+    signerAddress: SuiAddress,
+    t: PayTransaction
+  ): Promise<Base64DataBuffer> {
+    try {
+      const tx = {
+        Pay: {
+          input_coins: t.inputCoins,
```

Comment:
this needs to be an array of `SuiObjectRef`.

can do something like:
```
const inputCoinRefs = await Promise.all(t.inputCoins.map((coin) => this.provider.getObjectRef(coin)));

const tx = {
  Pay: {
    input_coins: inputCoinRefs,
    ...
```

Reply:
alternatively, since this is `local-txn-data-serializer`, you should consider letting users input the entire `SuiObjectRef` themselves instead of making an extra API request